EditVR/contenta_jsonapi
# FKED-74 [Contenta Drupal install] Example content

## [FKED-74 Contenta Drupal install - Example content](https://fourkitchens.atlassian.net/browse/FKED-74)

### Description of Work:
The contenta distro provides a framework for headless Drupal sites but does not enforce a specific content model. This changeset creates ‘experience’, ‘scene’, and ‘component’ nodes as sample content for the EditVR project. The demo content may be added to the site on installation. This work is largely based on the contenta recipe_magazin sample-content migration module.

### This PR does the following:
- Includes a new editvr_sample_content module within the contenta profile.
- Defines four content migration plugins (Experiences, Scenes, Components, and Files.)
- Defines two content migration source plugins (CSV and Directory)
- Includes 3 CSV files for storing migration data.
- Includes 9 file assets (audio and images) to attach to demo content.
- Defines a MigrationRunner service for executing migrations.
- Includes module install and module uninstall hooks to automatically execute migration tasks.
- Defines an EditVR optional module plugin to allow sample content to appear in the site-installation process and alters the contenta optional module plugin to hide the recipe_magazin content option.

### Test Instructions

- [x] Setup the local environment
  - [x] Checkout the EditVR-Project repo pointing to this branch. `composer create-project EditVR/editvr-project:dev-fked-74 --repository='{"type":"vcs","url":"https://github.com/EditVR/editvr-project"}' FKED-74`
  - [x] Create a new project with lando `lando init -r drupal8`
  - [x] Start web services `lando start`

- [x] Install the Drupal distro (see known issues below)
  - [x] Navigate to the new website to begin the install process. (http://editvr.lndo.site)
  - [x] On the "Configure Site" install task, set any valid username, password, and email to proceed with the installation.
  - [x] On the "Install and configure modules" task...
    - [x] Verify that "EditVR Sample Content" is listed in the module list. A description appears in the element. Additional installation notes appear below the element in a collapsible fieldset.
    - [x] Verify that the input is selected by default for the "EditVR Sample Content".
    - [x] Verify that the contenta "Magazin" demo content module does not appear.
    - [x] Click "Save and Continue" to complete the installation.
  - [x] Review the status area to confirm two success messages: 
      1. "The migration for demo content was successful."
      2. "Congratulations, you installed Contenta JSON API!"

- [x] Navigate to the Content Overview interface (/admin/content)
  - [x] Verify that 6 nodes were created through the installation process.
  - [x] Confirm that all data and file assets were correctly defined for each node. Fields are empty or in their default state unless noted below:
    - [x] Title: Drupalcon Demo, Content Type: Experience
      * Two scenes are attached, "Mike's Workshop" and "Mike's Office"
      * Button OnEnter Sound: audio-enter.wav (verify file is attached)
      * Button OnClick Sound: audio-click.wav (verify file is attached)
      * Info Button: Image of lowercase "i" in a circle
      * Navigation Button: Image of Footprints
      * Sound Button: Image of Speaker
      * Experience URL: "https://fourkitchens.github.io/webvr-studio/"
      * Experience Path: "drupalcon"
    - [x] Title: Mike's Workshop, Content Type: Scene
      * Photosphere: photosphere-mike-workshop.jpg (verify image is attached)
      * Slug: "mike-workshop"
    - [x] Title: Mike's Office, Content Type: Scene
      * Photosphere: photosphere-mike-office.jpg (verify image is attached)
      * Components: "Tabletop Gamers", "Change the World...", "Mike's Workshop"
      * Slug: "mike-office"
    - [x] Title: Tabletop Gamers, Content Type: Component
      * Type: Panel Image
      * Image: component-mike-greeting.png (verify image is attached)
      * RotationX: 0, RotationY: 90, X:4.51126 , Y: 1.20365, Z: -2.27527
    - [x] Title: Change the World..., Content Type: Component
      * Type: Panel Image
      * Image: component-mike-poster.png (verify image is attached)
      * RotationX: 0, RotationY: 90, X:3.41178, Y: 1.05444, Z: 2.33171
    - [x] Title: Mike's Workshop, Content Type: Component
      * Type: Link
      * RotationX: 0, RotationY: 90, X:5.05824, Y: -1.72253, Z: 2.70399

### Known Issues
* The contenta project currently throws errors on installation of the core/media modules. These errors are related to a file permissions issue that has been documented in another ticket FKED-76.
* The project also throws error related to expected public and private keys. This issue is unrelated to this work.
